### DISCUSSION TITLE
Best model and settings for (not clean) audiodramas?

### DISCUSSION THREAD

**Discussion by Pocokk** [openai/whisper#1928] [Q&A]
Hey there!

As my hearing is not so great, the voices usually get lost amidst of various sound effect (gunshot, street noise, background distortions etc) and I don't understand the context of certain audiodramas, its just a big mess for me sometimes.

That's why I'm asking  here and now the community's help, which model and settings should I use for those type of transcribing, when the source is not clean, it includes a lot of extra sound effects and overall can be messy to hear out words even for a mother-tongued English speaking person in the first place.

Has anyone figured out the optimal usage in that case?

I appreciate all the help here and thank you so much in advance!

**Comment by glangford** [comment]
If you haven't had success transcribing noisy audio with the large-v2 model, using a tool such as `demucs` to separate the spoken portion prior to running whisper might be a good first step. 

You can use a command similar to this to separate vocals and 'everything else':
`python -m demucs --two-stems vocals --mp3 -o demucs noisy.mp4`
(*The --two-stems=vocals option allows separating vocals from the rest of the accompaniment (i.e., karaoke mode)*)

Spleeter is another option (I haven't used that).

demucs is headlined as providing "music source separation", I don't know how well it would perform on your particular audio with FX noise so it's worth experimenting.

See also
* https://github.com/openai/whisper/discussions/1833
* https://github.com/openai/whisper/discussions/1907#discussioncomment-7888173

https://github.com/adefossez/demucs

**Reply by Pocokk** [reply]
Well, the first try was a disaster: I set it up, it worked until 100% progress then suddenly the whole system frozen with maximum CPU/HDD load, here is the photo I took 20 minutes later. As there wasn't anything in the (Windows) Event Viewer logs, I can't provide a source about what went wrong, but now I'm bit reluctant to try again. I've also looked it up and couldn't find a related issue in the past, which is a bit disturbing. 

![20231231_104042 1](https://github.com/openai/whisper/assets/81096046/29a146d2-2293-4437-96b3-eef738cbd11c)

**Reply by phineas-pta** [reply]
demucs has tendency to use a lot of ram (not vram) which freeze pc with not enough ram

anyway demucs and others like spleeter are for dealing with songs, so sound effect in your case is not suitable, there's no automatic solution for now, only manual editing

**Reply by Pocokk** [reply]
That's sad to read, thank you.

**Reply by glangford** [reply]
fyi

* https://github.com/facebookresearch/demucs/issues/498
